Communication updated

ninjacoderrakesh opened the pull request:
- Worked on updated communication methods

ninjacoderrakesh committed:
Add listner while calling startcamera And Update Readme file

ninjacoderrakesh committed:
Resolve Api 'Bindings' Issues.

Solved communication issues with adding request queue and await for callback.

ninjacoderrakesh committed:
Resolve Api 'Bindings' Issues.& Update README.md file

ninjacoderrakesh committed:
Resolve Api 'Bindings' Issues

ninjacoderrakesh committed:
Resolve communication method names issue.

ninjacoderrakesh committed:
Removed SDK code

ninjacoderrakesh committed:
Add new four calling methods of designhub SDK communication"

implemented new three methods startEyerwear,Load Variation,switch Context,Take screenshot.

ninjacoderrakesh committed:
Updated README file

nickjtch commented on README.md:
```diff
@@ -128,41 +130,138 @@ public class MainActivity extends AppCompatActivity implements WebviewListener{
+      public void onProgressCallback(String action) {
+      }
+
+      @Override
+      public void onTrackingCallback(String action) {
+
+      }
```
Load Variation doesn't have a trackingCallback

nickjtch commented on README.md:
```diff
@@ -128,41 +130,138 @@ public class MainActivity extends AppCompatActivity implements WebviewListener{
+      public void onProgressCallback(String action) {
+      }
+
+      @Override
+      public void onTrackingCallback(String action) {
+         progressDialog.setMessage("" + action);
+      }
```
SwitchContext doesn't have a tracking callback

nickjtch commented on README.md:
```diff
@@ -128,41 +130,138 @@ public class MainActivity extends AppCompatActivity implements WebviewListener{
+                dialog.show();
+      }
+
+      @Override
+      public void onProgressCallback(String action) {
+      }
+
+      @Override
+      public void onTrackingCallback(String action) {
+         progressDialog.setMessage("" + action);
+      }
```
Take screenshot doesnt have progress + tracking callbacks

nickjtch commented on README.md:
````diff
@@ -128,41 +130,138 @@ public class MainActivity extends AppCompatActivity implements WebviewListener{
 ```java
-designhubzVar.startCamera();
+progressDialog.show();
+designhubzVar.startEyewearTryon(new OnEyewearRequestCallback() {
````
This should take also as an input the Id of the eyewear

nickjtch commented on README.md:
````diff
@@ -128,41 +130,138 @@ public class MainActivity extends AppCompatActivity implements WebviewListener{
 ```java
-designhubzVar.getProduct();
+ progressDialog.show();
+ designhubzVar.loadVariation(new OnEyewearRequestCallback() {
````
This should take as an input the Id of the eyewear

ninjacoderrakesh committed:
Resolve PR #2

ninjacoderrakesh committed:
Update README File

ninjacoderrakesh committed:
Cleanup code and handle callback from SDK

1.Receive screenshot in bitmap and show with new method call showImageInDialog()
2.Handle callback result with specific datatype
3.Removed Unnecessory code.
4.Update README file.

ninjacoderrakesh committed:
Update README file

nickjtch commented on app/src/main/java/com/designhubzandroidexample/VideoViewActivity.java:
```diff
@@ -178,100 +103,149 @@ public void onClose(View view) {
+            }
 
+            @Override
+            public void onTrackingCallback(String message) {
```
Wouldn't it be better if we create an enum for this?

ninjacoderrakesh committed:
Added comments for callbacks and solved PR regarding onTrackingCallback

nickjtch commented on README.md:
```diff
@@ -145,18 +145,34 @@ DesignhubzWebview.initializeComponents(this);
 
       @Override
       public void onTrackingCallback(String message) {
+          // write your code to process or show tracking status
```
TrackingStatus instead of String

ninjacoderrakesh committed:
added trackingStatus instead of string

nickjtch committed:
Merge branch 'main' into communication-updated
